### ISSUE TITLE
Possible to use different fluid force configuration among different links?

### ISSUE THREAD

**Issue by Chuanfang-Neptune** [google-deepmind/mujoco#1676]
Hello,

I'm a phd student working on simulating amphibious animal/robot in MuJoCo. From the documentation on [fluid forces](https://mujoco.readthedocs.io/en/latest/computation/fluid.html#fluid-forces) I read that I could activate the built-in inertia or ellipsoid model by giving values to [density](https://mujoco.readthedocs.io/en/latest/XMLreference.html#option-density) and [viscosity](https://mujoco.readthedocs.io/en/latest/XMLreference.html#option-viscosity). 

I see that both these two variables are top-level elements rather than link specific elements, which means it works by affecting the solver and the same fluid force config would apply to the whole body. I wonder if it is possible to use different viscosity and density for different links of the body in order to simulate the transition state between liquid and air? 

Thanks!

**Comment by yuvaltassa** [comment]
The quick answer is "no".

However this a valid thing to want. The problem is more about how to define it. First, do you really want to define it according to body or geom? Isn't the thing you really want is to define density/viscosity in a volume of space? Like say above a plane it's something and below it something else. Or maybe you want it to be a general function of height to simulate air thinning?

Basically it's not clear how to expose it in a general way.

Swim in 100% water while you ponder it 😄

**Comment by Chuanfang-Neptune** [comment]
Thanks a lot for your reply!

Yes! I agree that the most ideal and general way is to define some volume-specific field parameters. But I would assume that a link-specific parameter would be something easier to implement. (I read that the way to implement a custom force field in MuJoCo is also to change mj_step, which affects the overall model performance) 

Maybe without an official implementation I would first try adding some volume-dependent parameter to the links and alter the external force to simulate similar effects.

Thanks!
